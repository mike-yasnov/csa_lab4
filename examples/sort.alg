// Program sort - simple sorting of three numbers

function main() {
    print("Sorting three numbers:");
    
    // Simplified version - sort 3 numbers
    var a = 30;
    var b = 10; 
    var c = 20;
    
    print("Before sorting:");
    print_number(a);
    print_number(b);
    print_number(c);
    
    // Simple sort with three comparisons
    if (a > b) {
        var temp = a;
        a = b;
        b = temp;
    }
    
    if (b > c) {
        var temp = b;
        b = c;
        c = temp;
    }
    
    if (a > b) {
        var temp = a;
        a = b;
        b = temp;
    }
    
    print("After sorting:");
    print_number(a);
    print_number(b);
    print_number(c);
}

main(); 