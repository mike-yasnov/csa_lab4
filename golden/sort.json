{
  "return_code": 0,
  "stdout": "Загрузка программы: /tmp/tempfile/program.bin\nЗагружено 100 инструкций\nЗагрузка данных: /tmp/tempfile/program_data.bin\nЗагружено 126 байт данных\nЗапуск выполнения (максимум 1000000 тактов)...\n\n=== РЕЗУЛЬТАТЫ ВЫПОЛНЕНИЯ ===\nСостояние: halted\nВыполнено инструкций: 3\nЗатрачено тактов: 8\nФинальный PC: 2\n\nВЫВОД ПРОГРАММЫ:\n[208][161][208][190][209][128][209][130][208][184][209][128][208][190][208][178][208][186][208][176] [209][130][209][128][209][145][209][133] [209][135][208][184][209][129][208][181][208][187]:\n\nПрограмма завершена успешно.\n",
  "stderr": "",
  "exec_log": "Cycle 000002: PC=0000 00(000000) Stack[0] TOS=empty\nCycle 000007: PC=0001 61(000001) Stack[1] TOS=0\nCycle 000008: PC=0002 01(000000) Stack[0] TOS=empty",
  "debug_listing": "INSTRUCTION MEMORY:\nAddress - Hex Code - Mnemonic\n----------------------------------------\n0000 - 00000000 - PUSH\n0002 - 61010000 - OUT 1\n0004 - 01000000 - POP\n0005 - 001E0000 - PUSH 30\n0007 - 002A0000 - PUSH 42\n0009 - 51000000 - STORE\n000A - 000A0000 - PUSH 10\n000C - 002E0000 - PUSH 46\n000E - 51000000 - STORE\n000F - 00140000 - PUSH 20\n0011 - 00320000 - PUSH 50\n0013 - 51000000 - STORE\n0014 - 00360000 - PUSH 54\n0016 - 61010000 - OUT 1\n0018 - 01000000 - POP\n0019 - 002A0000 - PUSH 42\n001B - 50000000 - LOAD\n001C - 61010000 - OUT 1\n001E - 01000000 - POP\n001F - 002E0000 - PUSH 46\n0021 - 50000000 - LOAD\n0022 - 61010000 - OUT 1\n0024 - 01000000 - POP\n0025 - 00320000 - PUSH 50\n0027 - 50000000 - LOAD\n0028 - 61010000 - OUT 1\n002A - 01000000 - POP\n002B - 002A0000 - PUSH 42\n002D - 50000000 - LOAD\n002E - 002E0000 - PUSH 46\n0030 - 50000000 - LOAD\n0031 - 34000000 - GT\n0032 - 412D0000 - JZ 45\n0034 - 002A0000 - PUSH 42\n0036 - 50000000 - LOAD\n0037 - 00510000 - PUSH 81\n0039 - 51000000 - STORE\n003A - 002E0000 - PUSH 46\n003C - 50000000 - LOAD\n003D - 002A0000 - PUSH 42\n003F - 51000000 - STORE\n0040 - 00510000 - PUSH 81\n0042 - 50000000 - LOAD\n0043 - 002E0000 - PUSH 46\n0045 - 51000000 - STORE\n0046 - 002E0000 - PUSH 46\n0048 - 50000000 - LOAD\n0049 - 00320000 - PUSH 50\n004B - 50000000 - LOAD\n004C - 34000000 - GT\n004D - 413F0000 - JZ 63\n004F - 002E0000 - PUSH 46\n0051 - 50000000 - LOAD\n0052 - 00550000 - PUSH 85\n0054 - 51000000 - STORE\n0055 - 00320000 - PUSH 50\n0057 - 50000000 - LOAD\n0058 - 002E0000 - PUSH 46\n005A - 51000000 - STORE\n005B - 00550000 - PUSH 85\n005D - 50000000 - LOAD\n005E - 00320000 - PUSH 50\n0060 - 51000000 - STORE\n0061 - 002A0000 - PUSH 42\n0063 - 50000000 - LOAD\n0064 - 002E0000 - PUSH 46\n0066 - 50000000 - LOAD\n0067 - 34000000 - GT\n0068 - 41510000 - JZ 81\n006A - 002A0000 - PUSH 42\n006C - 50000000 - LOAD\n006D - 00590000 - PUSH 89\n006F - 51000000 - STORE\n0070 - 002E0000 - PUSH 46\n0072 - 50000000 - LOAD\n0073 - 002A0000 - PUSH 42\n0075 - 51000000 - STORE\n0076 - 00590000 - PUSH 89\n0078 - 50000000 - LOAD\n0079 - 002E0000 - PUSH 46\n007B - 51000000 - STORE\n007C - 005D0000 - PUSH 93\n007E - 61010000 - OUT 1\n0080 - 01000000 - POP\n0081 - 002A0000 - PUSH 42\n0083 - 50000000 - LOAD\n0084 - 61010000 - OUT 1\n0086 - 01000000 - POP\n0087 - 002E0000 - PUSH 46\n0089 - 50000000 - LOAD\n008A - 61010000 - OUT 1\n008C - 01000000 - POP\n008D - 00320000 - PUSH 50\n008F - 50000000 - LOAD\n0090 - 61010000 - OUT 1\n0092 - 01000000 - POP\n0093 - 44000000 - RET\n0094 - 43000000 - CALL\n0096 - 01000000 - POP\n0097 - 70000000 - HALT\n\nDATA MEMORY:\nAddress - Hex Dump - ASCII\n----------------------------------------\n0000 - D0 A1 D0 BE D1 80 D1 82 D0 B8 D1 80 D0 BE D0 B2  - ................\n0010 - D0 BA D0 B0 20 D1 82 D1 80 D1 91 D1 85 20 D1 87  - .... ........ ..\n0020 - D0 B8 D1 81 D0 B5 D0 BB 3A 00 00 00 00 00 00 00  - ........:.......\n0030 - 00 00 00 00 00 00 D0 94 D0 BE 20 D1 81 D0 BE D1  - .......... .....\n0040 - 80 D1 82 D0 B8 D1 80 D0 BE D0 B2 D0 BA D0 B8 3A  - ...............:\n0050 - 00 00 00 00 00 00 00 00 00 00 00 00 00 D0 9F D0  - ................\n0060 - BE D1 81 D0 BB D0 B5 20 D1 81 D0 BE D1 80 D1 82  - ....... ........\n0070 - D0 B8 D1 80 D0 BE D0 B2 D0 BA D0 B8 3A 00        - ............:.\n"
}